tests/test_generator.py

```diff
@@ -255,55 +243,110 @@ def test_mkdir_file(generator_instance: Generator, test_dir: str):
+    )
+    list_of_files_strs = sorted(glob(path_to_files + "/*"))
+    yield [pathlib.Path(str_path) for str_path in list_of_files_strs]
 
 
-def test_copy_directory(get_gen_copy_dir, generator_instance: Generator):
-    """Test Generator._copy_files helper function with directory"""
-    files = EntityFiles(copy=get_gen_copy_dir)
-    cmd_list = generator_instance._copy_files(files, generator_instance.root)
-    assert isinstance(cmd_list, CommandList)
-    assert len(cmd_list) == 1
-    assert (
-        str(generator_instance.root)
-        and get_gen_copy_dir in cmd_list.commands[0].command
+@pytest.fixture
+def directory(fileutils):
+    directory = fileutils.get_test_conf_path(
+        osp.join("generator_files", "easy", "correct/")
     )
+    yield [pathlib.Path(directory)]
 
 
-def test_symlink_file(get_gen_symlink_dir, generator_instance: Generator):
-    """Test Generator._symlink_files helper function with file list"""
-    symlink_files = sorted(glob(get_gen_symlink_dir + "/*"))
-    files = EntityFiles(symlink=symlink_files)
-    cmd_list = generator_instance._symlink_files(files, generator_instance.root)
+@pytest.fixture
+def source(request, files, directory):
+    if request.param == "files":
+        return files
+    elif request.param == "directory":
+        return directory
+
+
+@pytest.mark.parametrize(
+    "dest",
+    (
+        pytest.param(None, id="dest as None"),
+        pytest.param(
+            pathlib.Path("/absolute/path"),
+            id="dest as valid path",
+        ),
+    ),
+)
+@pytest.mark.parametrize("source", ["files", "directory"], indirect=True)
+def test_copy_files_valid_dest(
+    dest, source, generator_instance: Generator, test_dir: str
+):
+    to_copy = [CopyOperation(src=file, dest=dest) for file in source]
+    gen = GenerationContext(pathlib.Path(test_dir))
+    cmd_list = generator_instance._copy_files(files=to_copy, context=gen)
     assert isinstance(cmd_list, CommandList)
-    for file, cmd in zip(symlink_files, cmd_list):
-        assert file in cmd.command
-
-
-def test_symlink_directory(generator_instance: Generator, get_gen_symlink_dir):
-    """Test Generator._symlink_files helper function with directory"""
-    files = EntityFiles(symlink=get_gen_symlink_dir)
-    cmd_list = generator_instance._symlink_files(files, generator_instance.root)
-    symlinked_folder = generator_instance.root / os.path.basename(get_gen_symlink_dir)
+    # Extract file paths from commands
+    cmd_src_paths = set()
+    for cmd in cmd_list.commands:
+        # Assert destination path exists in cmd
+        assert create_final_dest(pathlib.Path(test_dir), dest) in cmd.command
+        src_index = cmd.command.index("copy") + 1
+        cmd_src_paths.add(cmd.command[src_index])
+    # Assert all file paths are in the command list
+    file_paths = {str(file) for file in source}
+    assert file_paths.issubset(
+        cmd_src_paths
+    ), "Not all file paths are in the command list"
```

Comment:
Should these sets be exactly the same?
```suggestion
    assert file_paths == cmd_src_paths, "Not all file paths are in the command list"
```

Reply:
yes, we don't manipulate the src paths, I am comparing that the list of src paths passed in the beginning match the source paths in the commandlist

Reply:
Gotcha! I was just a little confused on if the `issubset` usage was an intentional choice and we expect potentially _more_ items to appear in `cmd_src_paths`. Looks like these tests have been updated to use `==` instead!

Marking as resolved!